Bug: block-scoped-var rule incorrectly flags break/continue with label

Currently, the `block-scoped-var` rule generates false warnings for `break` and `continue` statements that use labels.

**Version:** v0.24.1

**Source:**
```js
/* eslint no-labels:0, block-scoped-var:2 */

foo:
for (var i = 0; i < 10; ++i) {
    bar:
    for (var j = 0; j < 13; ++j) {
        if (j === 7) {
            continue foo;  // warning
        }

        if (j === 2) {
            break bar;  // warning
        }
    }
}
```

**Config:** Invoked with `eslint --no-eslintrc --reset`, so the complete config is in the source config.

**Output:**
```
   8:21  error  "foo" used outside of binding context  block-scoped-var
  12:18  error  "bar" used outside of binding context  block-scoped-var
```

--- a/lib/rules/block-scoped-var.js
+++ b/lib/rules/block-scoped-var.js
@@ -320,7 +320,7 @@ module.exports = function(context) {
 
         "Identifier": function(node) {
             var ancestor = context.getAncestors().pop();
-            if (isDeclaration(node, ancestor) || isProperty(node, ancestor) || ancestor.type === "LabeledStatement") {
+            if (isDeclaration(node, ancestor) || isProperty(node, ancestor) || ancestor.type === "LabeledStatement" || ancestor.type === "BreakStatement" || ancestor.type === "ContinueStatement") {
                 return;
             }
 

--- a/tests/lib/rules/block-scoped-var.js
+++ b/tests/lib/rules/block-scoped-var.js
@@ -60,6 +60,8 @@ eslintTester.addRuleTest("lib/rules/block-scoped-var", {
         "function f(){ for(var a in {}) a; }",
         "function f(){ switch(2) { case 1: var b = 2; b; break; default: b; break;} b; }",
         "a:;",
+        "foo: while (true) { bar: for (var i = 0; i < 13; ++i) {if (i === 7) break foo; } }",
+        "foo: while (true) { bar: for (var i = 0; i < 13; ++i) {if (i === 7) continue foo; } }",
         { code: "const React = require(\"react/addons\");const cx = React.addons.classSet;", globals: { require: false }, ecmaFeatures: { globalReturn: true, modules: true, blockBindings: true }},
         { code: "var v = 1;  function x() { return v; };", ecmaFeatures: { globalReturn: true }},
         { code: "import * as y from \"./other.js\"; y();", ecmaFeatures: { modules: true }},

